src/plette/lockfiles.py

```diff
@@ -64,99 +52,126 @@ def _copy_jsonsafe(value):
+            else:
+                PackageCollection.validate(value)
 
     @classmethod
-    def load(cls, fh, encoding=None):
+    def load(cls, f, encoding=None):
         if encoding is None:
-            data = json.load(fh)
+            data = json.load(f)
         else:
-            data = json.loads(fh.read().decode(encoding))
-        return cls(**data)
+            data = json.loads(f.read().decode(encoding))
+        return cls(data)
 
     @classmethod
     def with_meta_from(cls, pipfile, categories=None):
         data = {
             "_meta": {
-                "hash": pipfile.get_hash().__dict__,
+                "hash": _copy_jsonsafe(pipfile.get_hash()._data),
                 "pipfile-spec": PIPFILE_SPEC_CURRENT,
-                "requires": _copy_jsonsafe(getattr(pipfile, "requires", {})),
+                "requires": _copy_jsonsafe(pipfile._data.get("requires", {})),
+                "sources": _copy_jsonsafe(pipfile.sources._data),
             },
         }
-
-        data["_meta"].update(asdict(pipfile.sources))
-
         if categories is None:
-            data["default"] = _copy_jsonsafe(getattr(pipfile, "packages", {}))
-            data["develop"] = _copy_jsonsafe(getattr(pipfile, "dev-packages", {}))
+            data["default"] = _copy_jsonsafe(pipfile._data.get("packages", {}))
+            data["develop"] = _copy_jsonsafe(pipfile._data.get("dev-packages", {}))
         else:
             for category in categories:
-                if category in ["default", "packages"]:
-                    data["default"] = _copy_jsonsafe(getattr(pipfile,"packages", {}))
-                elif category in ["develop", "dev-packages"]:
-                    data["develop"] = _copy_jsonsafe(
-                            getattr(pipfile,"dev-packages", {}))
+                if category == "default" or category == "packages":
+                    data["default"] = _copy_jsonsafe(pipfile._data.get("packages", {}))
+                elif category == "develop" or category == "dev-packages":
+                    data["develop"] = _copy_jsonsafe(pipfile._data.get("dev-packages", {}))
                 else:
-                    data[category] = _copy_jsonsafe(getattr(pipfile, category, {}))
+                    data[category] = _copy_jsonsafe(pipfile._data.get(category, {}))
         if "default" not in data:
-            data["default"] = {}
+            data["default"]  = {}
         if "develop" not in data:
             data["develop"] = {}
         return cls(data)
 
     def __getitem__(self, key):
-        value = self[key]
+        value = self._data[key]
         try:
             if key == "_meta":
-                return Meta(**value)
-            return PackageCollection(value)
+                return Meta(value)
+            else:
+                return PackageCollection(value)
         except KeyError:
             return value
 
+    def __setitem__(self, key, value):
+        if isinstance(value, DataView):
```

Comment:
I think you've missed the `DataView` import

Reply:
Thanks!

Reply:
I release 2.0.1.

Reply:
I don't think the fix ended up in the new release. I can see the bug still present in the new `2.0.1` package `https://files.pythonhosted.org/packages/94/68/d5c6083ef944d56452dd10858b5c59da68c6eeae2a4953c1fca1bfcfd8e3/plette-2.0.1.tar.gz`

Reply:
Are you sure you look in the correct place?

Reply:
I got mislead as you've fixed one occurrence of the bug but not all of them. In the codebase there are still few `DataView` references https://github.com/search?q=repo%3Asarugaku%2Fplette%20DataView&type=code

Reply:
Yes, that is correct (sigh ...) there are two more...

Reply:
Should be finally fixed with 2.0.2. Apologies for that.

Reply:
> Should be finally fixed with 2.0.2. Apologies for that.

No problem. Unsolicited suggestion: both the initial bug report, which was introduced by a code regression, and the import issues of this PR, should be covered by tests and static code analysis.

Reply:
The tests suite is  unfortunately still very weak.
What static analysis tool would you suggest here?
My editor (vscode) gives already tons of warning, so some are easy to miss...